addons/sales_team/models/crm_team.py

```diff
@@ -63,6 +67,29 @@ def _get_default_favorite_user_ids(self):
     dashboard_button_name = fields.Char(string="Dashboard Button", compute='_compute_dashboard_button_name')
     dashboard_graph_data = fields.Text(compute='_compute_dashboard_graph')
 
+    @api.depends('crm_team_member_ids.active')
+    def _compute_member_ids(self):
+        for team in self:
+            team.member_ids = team.mapped('crm_team_member_ids.user_id').ids
+
+    def _inverse_member_ids(self):
+        for team in self:
+            memberships = team.crm_team_member_all_ids
+            # compute diff: users to add, memberships to remove or to activate again
+            users_new = team.member_ids - memberships.mapped('user_id')
+            memberships_unlink = memberships.filtered(lambda m: m.user_id not in team.member_ids)
+            memberships_activate = memberships.filtered(lambda m: m.user_id in users_new and not m.active)
+
+            team.crm_team_member_all_ids = [(0, 0, {
+                'user_id': user.id,
+                'active': True,
+            }) for user in users_new] + [
+                (2, m.id) for m in memberships_unlink
+            ] + [(1, m.id, {'active': True}) for m in memberships_activate]
```

Comment:
Shouldn't you *deactivate* memberships instead of deleting them?
Assuming this is the case, I would have written it this way:
```suggestion
        for team in self:
            memberships = team.crm_team_member_all_ids
            # add missing memberships
            memberships.create([
                {'crm_team_id': team.id, 'user_id': user.id}
                for user in team.member_ids - memberships.user_id
            ])
            # activate or deactivate other memberships depending on members
            for membership in memberships:
                membership.active = membership.user_id in team.member_ids
```
The `create` part is batched by team, and the assignment of `active` is eventually done in batch by the ORM, too.

Reply:
Thinking about it it seems like an interesting idea. Especially you may switch in mono-membership / multi membership mode, archiving is interesting.